plugins/CoreAdminHome/Commands/FixDuplicateLogActions.php

```diff
@@ -0,0 +1,215 @@
+use Piwik\DataAccess\Actions;
+use Piwik\DataAccess\ArchiveInvalidator;
+use Piwik\Plugin\ConsoleCommand;
+use Piwik\Plugins\CoreAdminHome\Model\DuplicateActionRemover;
+use Piwik\Timer;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * Finds duplicate actions rows in log_action and removes them. Fixes references to duplicate
+ * actions in the log_link_visit_action table, log_conversion table, and log_conversion_item
+ * table.
+ *
+ * Prior to version 2.11, there was a race condition in the tracker where it was possible for
+ * two or more actions with the same name and type to be inserted simultaneously. This resulted
+ * in inaccurate data. A Piwik database with this problem can be fixed using this class.
+ *
+ * With version 2.11 and above, it is still possible for duplicate actions to be inserted, but
+ * ONLY if the tracker's PHP process fails suddenly right after inserting an action. This is
+ * very rare, and even if it does happen, report data will not be affected, but the extra
+ * actions can be deleted w/ this class.
+ */
+class FixDuplicateLogActions extends ConsoleCommand
+{
+    /**
+     * Used to invalidate archives. Only used if $shouldInvalidateArchives is true.
+     *
+     * @var ArchiveInvalidator
+     */
+    private $archiveInvalidator;
+
+    /**
+     * DAO used to find duplicate actions in log_action and fix references to them in other tables.
+     *
+     * @var DuplicateActionRemover
+     */
+    private $duplicateActionRemover;
+
+    /**
+     * DAO used to remove actions from the log_action table.
+     *
+     * @var Actions
+     */
+    private $actionsAccess;
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * Constructor.
+     *
+     * @param ArchiveInvalidator $invalidator
+     * @param DuplicateActionRemover $duplicateActionRemover
+     * @param Actions $actionsAccess
+     * @param LoggerInterface $logger
+     */
+    public function __construct(ArchiveInvalidator $invalidator = null, DuplicateActionRemover $duplicateActionRemover = null,
+                                Actions $actionsAccess = null, LoggerInterface $logger = null)
+    {
+        parent::__construct();
+
+        if ($invalidator === null) {
+            $invalidator = new ArchiveInvalidator();
+        }
+        $this->archiveInvalidator = $invalidator;
+
+        if ($duplicateActionRemover === null) {
+            $duplicateActionRemover = new DuplicateActionRemover();
+        }
+        $this->duplicateActionRemover = $duplicateActionRemover;
+
+        if ($actionsAccess === null) {
+            $actionsAccess = new Actions();
+        }
+        $this->actionsAccess = $actionsAccess;
+
+        if ($logger === null) {
+            $logger = StaticContainer::get('Psr\Log\LoggerInterface');
+        }
+        $this->logger = $logger;
```

Comment:
> You could also get all those dependencies from the container so that you can use dependency injection without having null in constructors

Would I have to access StaticContainer directly? Maybe it's possible to do it once for the console command? That would be nice.